Classes/common/CDTIncrementalStore.m

```diff
@@ -0,0 +1,2798 @@
+            NSString *bytes =
+                [self encodeBlob:data withName:name inStore:blobStore withMIMEType:mimeType];
+            return @{
+                name : bytes,
+                MakeMeta(name) : @{kCDTISMIMETypeKey : @"application/octet-stream"}
+            };
+        }
+        case NSTransformableAttributeType: {
+            NSString *xname = [attribute valueTransformerName];
+            NSString *mimeType = @"application/octet-stream";
+            Class myClass = NSClassFromString(xname);
+            // Yes, we could try/catch here.. but why?
+            if ([myClass respondsToSelector:@selector(MIMEType)]) {
+                mimeType = [myClass performSelector:@selector(MIMEType)];
+            }
+            id xform = [[myClass alloc] init];
+            // use reverseTransformedValue to come back
+            NSData *save = [xform transformedValue:obj];
+            NSString *bytes =
+                [self encodeBlob:save withName:name inStore:blobStore withMIMEType:mimeType];
+
+            return @{ name : bytes, MakeMeta(name) : @{kCDTISMIMETypeKey : mimeType} };
+        }
+        case NSObjectIDAttributeType: {
+            // I don't think converting to a ref is needed, besides we
+            // would need the entity id to decode.
+            NSManagedObjectID *oid = obj;
+            NSURL *uri = [oid URIRepresentation];
+            return @{
+                name : [uri absoluteString],
+            };
+        }
+        case NSDecimalAttributeType: {
+            NSDecimalNumber *dec = obj;
+            NSString *desc = [dec description];
+            NSDecimal val = [dec decimalValue];
+            NSData *data = [NSData dataWithBytes:&val length:sizeof(val)];
+            NSString *b64 = [data base64EncodedStringWithOptions:0];
+            NSMutableDictionary *meta = [NSMutableDictionary dictionary];
+            meta[kCDTISDecimalImageKey] = b64;
+
+            if ([dec isEqual:[NSDecimalNumber notANumber]]) {
+                meta[kCDTISFPNaNKey] = @"true";
+                desc = @"0";
+            }
+
+            return @{name : desc, MakeMeta(name) : [NSDictionary dictionaryWithDictionary:meta]};
+        }
+        case NSDoubleAttributeType: {
+            NSNumber *num = obj;
+            double dbl = [num doubleValue];
+            NSNumber *i64 = @(*(int64_t *)&dbl);
+            NSMutableDictionary *meta = [NSMutableDictionary dictionary];
+            meta[kCDTISDoubleImageKey] = i64;
+
+            if ([num isEqual:@(INFINITY)]) {
+                num = @(DBL_MAX);
+                meta[kCDTISFPInfinityKey] = @"true";
+            }
+            if ([num isEqual:@(-INFINITY)]) {
+                num = @(-DBL_MAX);
+                meta[kCDTISFPNegInfinityKey] = @"true";
+            }
+            if ([num isEqual:@(NAN)]) {
+                num = @(0);  // not sure what to do here
+                dbl = 0.;
+                meta[kCDTISFPNaNKey] = @"true";
+            }
```

Comment:
Why not just capture the full 64-bit value as a hex string?  This would give you a single means to capture plus/minus infinity and NAN (in all it's various forms!).